plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -111,8 +111,10 @@ private DeployPreferencesPanel getPreferencesPanel(Composite container) throws C
     IFacetedProject facetedProject = ProjectFacetsManager.create(project);
 
     if (AppEngineStandardFacet.hasAppEngineFacet(facetedProject)) {
+      setTitle(Messages.getString("standard.page.title"));
       return new StandardDeployPreferencesPanel(container, project, getLayoutChangedHandler());
     } else if (AppEngineFlexFacet.hasAppEngineFacet(facetedProject)) {
+      setTitle(Messages.getString("flex.page.title"));
       return new FlexDeployPreferencesPanel(container);
     } else {
```

Comment:
This shouldn't happen, but if it does is this really where we want to throw an exception? It feels safer to just set the title in this case.